### PR TITLE
Remove some unused/old files

### PR DESCRIPTION
**Pull request by MeshCollider**
Just a quick cleanup of a few files which are either unused now (due to recent changes) or never used (e.g. sample images). Also moves a stray post file into the correct directory.

### CODE REVIEW

File: _data/authors.yml [hunk 1]
```diff
@@ -1,41 +0,0 @@
-# Authors
```
Comment (MarcoFalke) [inline _data/authors.yml]: Those seem to be used here: `git grep 'author: ' ./_posts/`

**Comment by MarcoFalke** [comment]
Concept ACK

File: _data/authors.yml (hunk 1 above)
Comment (MeshCollider) [reply _data/authors.yml]: Oops, fixed. Thanks

**Comment by MarcoFalke** [comment]
needs rebase

**Comment by harding** [comment]
@MeshCollider this still needs rebase.  In addition, I'm pretty sure several of the removed files are used, such as `_data/source_navigation.txt` and the files in the `_releases` directory.

**Comment by MeshCollider** [comment]
Ah apologies, I will just close this

**Pull request closed without merging**